### DISCUSSION TITLE
Unable to send email from EME even though other smtp plugins are able to do so

### DISCUSSION THREAD

**Discussion by refugeindhamma** [liedekef/events-made-easy#52] [Q&A] [answered]
Hi Franky,

I'm unable to send emails from EME. When I test the same SMTP settings with other plugins, I'm able to send emails. I keep getting the error that "MAIL FROM command failed". I checked my user email, and the Email Settings and all have no problem.

![Screenshot 2023-01-31 at 19-29-19 Email management ‹ Refuge In Dhamma — WordPress](https://user-images.githubusercontent.com/101639308/215750086-267f0883-83cc-4032-943a-eb0bcd752aae.png)

My email service provider documentation:
![Screenshot 2023-01-31 at 19-42-09 Technical Setup Purelymail](https://user-images.githubusercontent.com/101639308/215751269-01a1cd2a-b84a-4f68-b710-4ebc020eec3a.png)

I have already checked that ports 465 & 587 are opened on my server and firewall. When I use port 465/TLS, nothing happens. No error message and no emails are sent. When I use port 587/TLS & 465/SSL, I get the "MAIL FROM command failed" error. When I use 587/SSL, there is also an error. 
![Screenshot 2023-01-31 at 19-50-37 Email management ‹ Refuge In Dhamma — WordPress](https://user-images.githubusercontent.com/101639308/215752437-ce70d095-b68f-417a-be48-6f01f8381626.png)

How to resolve this?

**Comment by liedekef** [comment]
The error seems to be in your sender config (empty from-address).
![image](https://user-images.githubusercontent.com/75667/215755358-5bb0456d-2d86-404d-97af-df913f7f8796.png)
In theory EME should take your blog admin account as sender+from email, unless that email is empty in which case it will fail. What are your settings there and does your blog admin have an email configured?

**Comment by refugeindhamma** [comment]
"Default email sender address" and "Default email sender name" are empty. I wanted to use the admin account info to send the email. And my admin user's email has the same URL as those set up at the email service provider, @dev.mydomain.com. 

In EME, I tried using the same sender + name as Fluent SMTP plugin but EME still fails whereas Fluent SMTP works.

**Comment by liedekef** [comment]
EME tries very hard to find a default contact person to use as "from". From the code:

```
                if ( function_exists( 'is_multisite' ) && is_multisite() ) {
                        $thisblog = get_current_blog_id();
                        $userinfo = get_user_by( 'email', get_blog_option( $thisblog, 'admin_email' ) );
                } else {
                        $userinfo = get_user_by( 'email', get_option( 'admin_email' ) );
                }
                // still nothing ? Can be if the main admin email is not matching a user, so get the first admin
                if ( ! $userinfo ) {
                        $eme_super_admins = get_super_admins();
                        if ( ! empty( $eme_super_admins ) ) {
                                $userinfo = get_user_by( 'login', $eme_super_admins[0] );
                        }
                }
```
The only condition is that the email uses as main admin also matches an actual WP user, otherwise it will fail.
Can it be you left "Default email sender address" and "Default email sender name" empty but activated "Force sender address everywhere" (the setting just underneath)? Then that will be the problem ... I advise to fill out a name and an email in the default settings (even if it is the email from your admin), then it should be ok.

The next version will take into account the "wrong" combo of these 3 settings (so if the setting "Force sender address everywhere" is active but "Default email sender address" and "Default email sender name" are empty, the from will be taken from the admin too).

**Comment by refugeindhamma** [comment]
"Force sender address everywhere" is turned off. I have also filled in "Default email sender address" & "Default email sender name" but it is still not working. I even changed email service provider and the same error occurs. 

Is it possible to use Fluent SMTP to send emails instead of EME? I tried using "Wordpress Email" and it failed too.

**Reply by liedekef** [reply]
No, we need to figure out why the "from" stays empty. If your admin email is not linked to a WP user, that will cause issues. Is that possibly the case currently?

**Reply by refugeindhamma** [reply]
> No, we need to figure out why the "from" stays empty. If your admin email is not linked to a WP user, that will cause issues. Is that possibly the case currently?

Nope. the email is linked to my admin user.

**Comment by refugeindhamma** [comment]
I managed to send the emails. It works only if I turn on "Force sender address everywhere". If it is turned off, even if my sender address is the same as that of the email service provider, it will give FROM error.

Edit: "Default email sender address" must be filled. If empty, it will give FROM error too.

**Reply by liedekef** [reply]
Yes, the default needs to be filled for the "force" to work correctly, that I'll try to fix for next time. I'll also try to be more strict on the "from", I'll rework the code on that later on. I'm still not totally clear why the default doesn't work for you. If you could give me some temp admin access on a test setup (where you can reproduce the problem), then I might find the cause more quickly and get a better fix in the code.

**Reply by refugeindhamma** [reply]
Sent the temp admin access link to your email.

**Comment by refugeindhamma** [comment, marked as answer]
For new users who face the same problem, the issue is resolved by checking your "General" page in Wordpress settings whether the email listed there is the same as your email platform's registered email. If it is different, then you need to change it to be the same.

Alternatively, if the email is different in the "General" page in Wordpress settings and you do not want to change it, you can put a different email (but still the same as your email platform's registered email) in the "Email" tab in EME settings and change "Force sender address everywhere" to "Yes". This way, Wordpress will be using the FROM email address in the "General" page and EME will be using the email address in the "Email" tab.

**Comment by liedekef** [comment]
Meanwhile the released version should be more robust and take the admin email and blog name as sender if nothing else matches.
